draft-irtf-cfrg-hybrid-kems.md

```diff
@@ -270,259 +296,352 @@ of testing):
+shared secret computation.  One additional hash function is required:
+
+* `KeyHash`: A hash function mapping byte strings of length `KEM_T.Nek +
+  KEM_PQ.Nek` to byte strings of length `KEM_H.Nss` (`CombineHash.Nh ==
+  KEM_H.Nss`)
 
-Similarly, `DeriveKeyPair` works as follows:
+The shared secret computation in `Encaps` and `Decaps` is then modified as
+follows:
 
 ~~~
-def DeriveKeyPair(seed):
-  (pq_DK, trad_DK, pq_EK, trad_EK) = expandDecapsulationKey(seed)
-  return dk, concat(pq_EK, trad_EK)
+ekh = KeyHash(concat(ek_T, ek_PQ))
+ss_H = CombinerHash(concat(ss_PQ, ss_T, ct_PQ, ct_T, ekh, label))
 ~~~
 
+### Instantiation with a Nominal Group
+
+The HashEverything and PreHashedKeys hybrid KEMs can be instantiated with a
+nominal group in place of the traditional KEM.  However, because the group
+construction used as a constituent does not comprise a secure KEM, these
+constructions are defined and analyzed separately.
 
-## 'Kitchen Sink' construction {#KitchenSink}
+To instantiate the HashEverything or PreHashedKeys hybrid KEM with a nominal
+group, the traditional KEM `KEM_T` is replaced with a "pseudo-KEM" `PseudoKEM_T`
+based on a nominal group `Group_T`.  We use the term "pseudo-KEM" because while
+this construction meets the KEM API definition, it is not a secure KEM.
 
-As indicated by the name, the `KitchenSink` puts 'the whole transcript'
-through the KDF. This relies on the minimum security properties of its
-component algorithms at the cost of more bytes needing to be processed by the
-KDF.
+The constants for the pseudo-KEM are defined as follows:
 
 ~~~
-def KitchenSink-KEM.SharedSecret(pq_SS, trad_SS, pq_CT, pq_EK, trad_CT,
-                                 trad_EK, label):
-    input = concat(pq_SS, trad_SS, pq_CT, pq_EK,
-                   trad_CT, trad_EK, label)
-    return KDF(input)
+PseudoKEM_T.Nseed = Group_T.Nseed
+PseudoKEM_T.Nek = GroupT.Nelem
+PseudoKEM_T.Ndk = GroupT.Nscalar
+PseudoKEM_T.Nct = GroupT.Nelem
+PseudoKEM_T.Nss = GroupT.Nss
+~~~
+
+The pseudo-KEM interface methods are defined as follows:
+
+~~~
+def PseudoGenerateKeyPair():
+    return DeriveKeyPair(random(Group_T.Nseed))
+
+def PseudoDeriveKeyPair(seed):
+    dk = Group_T.RandomScalar(seed)
+    ek = Group_T.Exp(Group_T.g, dk)
+    return (ek, dk)
+
+def PseudoEncaps(ek):
+    ct, sk_E = GenerateKeyPair()
+    ss = GroupT.ElementToSharedSecret(Group_T.Exp(ek, sk_E))
+    return (ct, ss)
+
+def PseudoDecaps(dk, ct):
+    return GroupT.ElementToSharedSecret(Group_T.Exp(ct, dk))
 ~~~
 
-### Security properties
+The HashEverything and PreHashedKeys hybrid KEMs are instantiated with a
+nominal group by replacing the `KEM_T` methods with the corresponding `Pseudo`
+methods.
 
-Because the entire hybrid KEM ciphertext and encapsulation key material are
-included in the KDF preimage, the `KitchenSink` construction is resilient
-against implementation errors in the component algorithms. <!-- TODO: cite
-that thing -->
+## HashTraditionalOnly
```

Comment:
I don't love this name (I don't quite understand what it's trying to say) and think `QSF` is as good a name as any, short, and unless you already know it's associated with X-Wing, non-controversial?

Reply:
@BTW, what it's trying to say is that the only metadata you hash in is on the T side.  🤷

Reply:
Let's leave it as-is and bike shed the name later. I think it's clear enough for now.

Reply:
Yes leaving it the existing document names as-is seems fine (QSF)